lib/rubocop/cop/lint/unreachable_code.rb

```diff
@@ -74,11 +74,21 @@ def flow_expression?(node)
         end
 
         def check_if(node)
+          return true if check_literal_if(node)
+
           if_branch = node.if_branch
           else_branch = node.else_branch
           if_branch && else_branch && flow_expression?(if_branch) && flow_expression?(else_branch)
         end
 
+        def check_literal_if(node)
+          return false unless flow_expression?(node.if_branch)
+
+          return true if !node.unless? && node.condition.truthy_literal?
+
+          node.unless? && node.condition.falsey_literal?
```

Comment:
I think you could easily extend this to also support `else` branches. Something like:

```ruby
def check_literal_if(node)
  branch = unconditional_branch(node)
  branch && flow_expression?(branch)
end

def unconditional_branch(node)
  return node.if_branch   if !node.unless? && node.condition.truthy_literal?
  return node.else_branch if node.unless?  && node.condition.falsey_literal?
  nil
end
```

This would catch cases like
```ruby
if false
  foo
else
  return
end
```